src/android/com/mady/wifi/api/WifiStatus.java

```diff
@@ -284,13 +286,39 @@ public boolean pingCmd(String addr) {
             int exit = pro.exitValue();
             if (exit == 0) {
                 return true;
-            } else {
-                //ip address is not reachable
-                return false;
             }
         } catch (IOException e) {
+            Log.d("RAVEN", "ICMP ping error - " + e.getMessage());
+        }
+        Log.d("RAVEN", "ICMP ping to " + addr + " failed, trying TCP connection over port 443");
+        if (isReachable(addr, 443, 5000)) return true;
+        Log.d("RAVEN", "TCP connection to  " + addr + " failed, trying TCP connection over port 80");
```

Comment:
These logs can be added at the beginning of the isReachable function instead. Also this specific one would be a duplication considering we already log on the isReachable if the TCP using 443 failed. 
```
        Log.d("RAVEN", "Trying TCP connection to " + addr + " over port " + openPort);
```